lib/pacemaker/pcmk_sched_actions.c

```diff
@@ -1375,88 +1375,113 @@ reschedule_recurring(pe_resource_t *rsc, const char *task, guint interval_ms,
     pe__set_action_flags(op, pe_action_reschedule);
 }
 
-gboolean
-check_action_definition(pe_resource_t * rsc, pe_node_t * active_node, xmlNode * xml_op,
-                        pe_working_set_t * data_set)
+/*!
+ * \internal
+ * \brief Handle any configuration change for an action
+ *
+ * Given an action from resource history, if the resource's configuration
+ * changed since the action was done, schedule any actions needed (restart,
+ * reload, unfencing, rescheduling recurring actions, etc.).
+ *
+ * \param[in] rsc     Resource that action is for
+ * \param[in] node    Node that action was on
+ * \param[in] xml_op  Action XML from resource history
+ *
+ * \return true if action configuration changed, otherwise false
+ */
+bool
+pcmk__check_action_config(pe_resource_t *rsc, pe_node_t *node, xmlNode *xml_op)
 {
     guint interval_ms = 0;
+    const char *task = NULL;
     const op_digest_cache_t *digest_data = NULL;
 
-    const char *task = crm_element_value(xml_op, XML_LRM_ATTR_TASK);
+    CRM_CHECK((rsc != NULL) && (node != NULL) && (xml_op != NULL),
+              return false);
 
-    CRM_CHECK(active_node != NULL, return FALSE);
+    task = crm_element_value(xml_op, XML_LRM_ATTR_TASK);
+    CRM_CHECK(task != NULL, return false);
 
     crm_element_value_ms(xml_op, XML_LRM_ATTR_INTERVAL_MS, &interval_ms);
+
+    // If this is a recurring action, check whether it has been orphaned
     if (interval_ms > 0) {
         if (action_in_config(rsc, task, interval_ms)) {
             pe_rsc_trace(rsc, "%s-interval %s for %s on %s is in configuration",
                          pcmk__readable_interval(interval_ms), task, rsc->id,
-                         active_node->details->uname);
-        } else if (pcmk_is_set(data_set->flags, pe_flag_stop_action_orphans)) {
+                         node->details->uname);
+        } else if (pcmk_is_set(rsc->cluster->flags,
+                               pe_flag_stop_action_orphans)) {
             schedule_cancel(rsc,
                             crm_element_value(xml_op, XML_LRM_ATTR_CALLID),
-                            task, interval_ms, active_node, "orphan");
-            return TRUE;
+                            task, interval_ms, node, "orphan");
+            return true;
         } else {
             pe_rsc_debug(rsc, "%s-interval %s for %s on %s is orphaned",
                          pcmk__readable_interval(interval_ms), task, rsc->id,
-                         active_node->details->uname);
-            return TRUE;
+                         node->details->uname);
+            return true;
         }
     }
 
-    crm_trace("Testing " PCMK__OP_FMT " on %s",
-              rsc->id, task, interval_ms, active_node->details->uname);
+    crm_trace("Checking %s-interval %s for %s on %s for configuration changes",
+              pcmk__readable_interval(interval_ms), task, rsc->id,
+              node->details->uname);
```

Comment:
This function call (or some very slight variation like changing the "for ..." string) has shown up in a couple patches now.  It might be nice to have a macro or inline function for it.

Reply:
Do you mean the "%s-interval %s for %s on %s" part?

Reply:
Yeah, the whole crm_trace call there was in several places.

Reply:
They're different enough to make it troublesome. A macro might help but I'm going to leave it for another PR